assets/js/custom/ProgramCredits.js

```diff
@@ -1,50 +1,196 @@
-    } else {
-      descriptionCreditsContainer.show()
-      descriptionHeadline.show()
-      editCreditsUI.addClass('d-none')
-      handleShowMore()
-    }
+  let languages = {}
+  let previousCreditsIndex = 0
+  let lastSavedCredits = credits.text().trim()
+
+  $(document).ready(function () {
+    getLanguages()
   })
 
-  $('#edit-credits-submit-button').on('click', () => {
+  function getLanguages () {
+    $.ajax({
+      url: '../languages',
+      type: 'get',
+      success: function (data) {
+        languages = data
+        populateSelector()
+      }
+    })
+  }
+
+  function populateSelector () {
+    creditLanguageSelectorList.empty()
+    creditLanguageSelectorList.append('<li class="mdc-list-item" data-value="default" role="option" tabindex="-1">' +
+        '<span class="mdc-list-item__ripple"></span>' +
+        '<span class="mdc-list-item__text">Default</span>' +
+        '</li>')
+
+    for (const language in languages) {
+      if (language.length <= 2) {
+        creditLanguageSelectorList.append(`<li class="mdc-list-item" data-value="${language}" role="option" tabindex="-1">\
+          <span class="mdc-list-item__ripple"></span>\
+          <span class="mdc-list-item__text">${languages[language]}</span>\
+          </li>`)
+      }
+    }
+
+    creditsSelect.layoutOptions()
+    resetCreditsEditor()
+  }
+
+  function getCustomTransalationSuccess (data) {
+    editCredits.val(data)
+    lastSavedCredits = data
+    editCredits.removeAttr('disabled')
+  }
+
+  function getCustomTransalationError () {
+    const defaultCredits = credits.text().trim()
+    editCredits.val(defaultCredits)
+    lastSavedCredits = defaultCredits
+    editCredits.removeAttr('disabled')
+  }
+
+  function getCustomTransaltion () {
+    editCredits.attr('disabled', '')
+
+    customTranslationApi.getCustomTransaltion(
+      programId,
+      creditsSelect.value,
+      getCustomTransalationSuccess,
+      getCustomTransalationError
+    )
+  }
+
+  function deleteCustomTranslationSuccess () {
+    snackbarNotification.show(translationDeleted, creditsSelectedText.text())
+    closeCreditsEditor()
+  }
+
+  function saveCustomTranslationSuccess () {
+    snackbarNotification.show(translationSaved, creditsSelectedText.text())
+    closeCreditsEditor()
+  }
+
+  function showCreditsError (error) {
+    editCredits.addClass('danger')
+    editCreditsError.show()
+    editCreditsError.text(error.message)
+  }
+
+  function saveCredits () {
     const newCredits = editCredits.val().trim()
     if (newCredits === credits.text().trim()) {
-      editCreditsUI.addClass('d-none')
-      descriptionHeadline.show()
-      descriptionCreditsContainer.show()
-      handleShowMore()
+      closeCreditsEditor()
       return
     }
 
-    const url = Routing.generate('edit_program_credits', { id: programId, new_credits: newCredits }, false)
+    const languageSelected = creditsSelect.value
+    if (languageSelected === '' || languageSelected === 'default') {
```

Comment:
I think it might be an empty string if it is still the default description or credit since the language selector does not show up in that case or if the language API still hasn't loaded for some reason. Either case, it is just another check to make sure it doesn't fall through the cases

Reply:
understood